index.js

```diff
@@ -1,2 +1,10 @@
 'use strict';
-module.exports = (cond, doIf, doElse) => val => cond ? doIf(val) : (doElse ? doElse(val) : val);
+
+module.exports = (cond, doIf, doElse) => val => {
+	if (typeof cond === 'function') {
+		return Promise.resolve(cond(val))
+			.then(bool => bool === true ? doIf(val) : (doElse ? doElse(val) : val));
+	}
+
+	return cond ? doIf(val) : (doElse ? doElse(val) : val);
```

Comment:
This needs to test for `true` too.

You also have the same logic in two places now, which could be DRY'd up into a function.